File: jambeth/jambeth-security/src/main/java/de/osthus/ambeth/privilege/factory/IEntityPrivilegeFactory.java
package de.osthus.ambeth.privilege.factory;

import de.osthus.ambeth.privilege.model.IPrivilege;
import de.osthus.ambeth.privilege.model.IPropertyPrivilege;

public interface IEntityPrivilegeFactory
{
	IPrivilege createPrivilege(boolean read, boolean create, boolean update, boolean delete, boolean execute, IPropertyPrivilege[] primitivePropertyPrivileges,
			IPropertyPrivilege[] relationPropertyPrivileges);
}

File: ambeth/Ambeth.Security/ambeth/privilege/factory/IEntityPrivilegeFactory.cs
using De.Osthus.Ambeth.Privilege.Model;

namespace De.Osthus.Ambeth.Privilege.Factory
{
    public interface IEntityPrivilegeFactory
    {
        IPrivilege CreatePrivilege(bool read, bool create, bool update, bool delete, bool execute, IPropertyPrivilege[] primitivePropertyPrivileges,
                IPropertyPrivilege[] relationPropertyPrivileges);
    }
}